src/edu/washington/escience/myria/operator/network/partition/HowPartitioned.java

```diff
@@ -0,0 +1,60 @@
+package edu.washington.escience.myria.operator.network.partition;
+
+import java.util.Set;
+
+import javax.annotation.Nullable;
+
+import jersey.repackaged.com.google.common.collect.ImmutableSet;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.primitives.Ints;
+
+/**
+ * How a dataset is partitioned. There are two fields being recorded for now, the partition function and the sequence of
+ * worker IDs. More information can be added in the future if needed.
+ */
+public class HowPartitioned {
```

Comment:
I think `how_partitioned` is not just about general partitioning strategy, it depends on what we put inside of it. Currently it has the function and the worker array, we can definitely add more if we want. PartitionInfo, on the other hand, sounds vague to me since "Info" doesn't indicate anything meaningful. 
Changing the name is not hard, but we need to remember to change the master catalog again.